### PR TITLE
Give better error messages when parsing fails

### PR DESCRIPTION
**Pull request by richvdh**
An attempt to give slightly more helpful error messages when we have a problem
parsing an event schema file (in particular, which file contains the problem,
and which property within it).

A bit of light refactoring to make it tractable.

### CODE REVIEW

**Comment by richvdh** [comment]
Handy hint: review the diff with w=1: https://github.com/matrix-org/matrix-doc/pull/339/files?w=1

File: templating/matrix_templates/units.py [hunk 1]
```diff
@@ -611,89 +634,100 @@ def load_event_schemas(self):
...
-                    #   ]
-                    # }
-                    ]
-                }
+            try:
+                schemata[filename] = self.read_event_schema(filepath)
+            except Exception, e:
+                e2 = Exception("Error reading event schema "+filepath+": "+
+                               str(e))
+                raise e2, None, sys.exc_info()[2]
```
Comment (Kegsay) [inline templating/matrix_templates/units.py]: What does this do? I know it throws but `sys.exc_info()[2]`? A comment here would be good.

**Comment by Kegsay** [comment]
LGTM aside from comment. We really need to just refactor this... all the recursion makes the logic hard to follow.

**Comment by richvdh** [comment]
> We really need to just refactor this... all the recursion makes the logic hard to follow.

I think it's getting better, very slowly? But yes, it is an unholy mess.

ptal, anyway

**Comment by Kegsay** [comment]
LGTM
